Geometry/HGCalCommonData/src/HGCalGeomParameters.cc

```diff
@@ -617,74 +675,128 @@ void HGCalGeomParameters::loadSpecParsHexagon8(const DDFilteredView& fv, HGCalPa
-                                << php.fracAreaMin_ << ":" << php.zMinForRad_;
-#endif
+
   php.radiusMixBoundary_ = DDVectorGetter::get("RadiusMixBoundary");
   std::for_each(php.radiusMixBoundary_.begin(), php.radiusMixBoundary_.end(), [](double& n) {
     n *= HGCalParameters::k_ScaleFromDDD;
   });
-#ifdef EDM_ML_DEBUG
-  for (unsigned int k = 0; k < php.radiusMixBoundary_.size(); ++k)
-    edm::LogVerbatim("HGCalGeom") << "HGCalParameters: Mix[" << k << "] R = " << php.radiusMixBoundary_[k];
-#endif
+
   php.slopeMin_ = getDDDArray("SlopeBottom", sv, 0);
   php.zFrontMin_ = getDDDArray("ZFrontBottom", sv, 0);
   std::for_each(php.zFrontMin_.begin(), php.zFrontMin_.end(), [](double& n) { n *= HGCalParameters::k_ScaleFromDDD; });
   php.rMinFront_ = getDDDArray("RMinFront", sv, 0);
   std::for_each(php.rMinFront_.begin(), php.rMinFront_.end(), [](double& n) { n *= HGCalParameters::k_ScaleFromDDD; });
-#ifdef EDM_ML_DEBUG
-  for (unsigned int k = 0; k < php.zFrontMin_.size(); ++k)
-    edm::LogVerbatim("HGCalGeom") << "HGCalParameters: Boundary[" << k << "] Bottom Z = " << php.zFrontMin_[k]
-                                  << " Slope = " << php.slopeMin_[k] << " rMax = " << php.rMinFront_[k];
-#endif
+
   php.slopeTop_ = getDDDArray("SlopeTop", sv, 0);
   php.zFrontTop_ = getDDDArray("ZFrontTop", sv, 0);
   std::for_each(php.zFrontTop_.begin(), php.zFrontTop_.end(), [](double& n) { n *= HGCalParameters::k_ScaleFromDDD; });
   php.rMaxFront_ = getDDDArray("RMaxFront", sv, 0);
   std::for_each(php.rMaxFront_.begin(), php.rMaxFront_.end(), [](double& n) { n *= HGCalParameters::k_ScaleFromDDD; });
-#ifdef EDM_ML_DEBUG
-  for (unsigned int k = 0; k < php.zFrontTop_.size(); ++k)
-    edm::LogVerbatim("HGCalGeom") << "HGCalParameters: Boundary[" << k << "] Top Z = " << php.zFrontTop_[k]
-                                  << " Slope = " << php.slopeTop_[k] << " rMax = " << php.rMaxFront_[k];
-#endif
+
   php.zRanges_ = DDVectorGetter::get("ZRanges");
   std::for_each(php.zRanges_.begin(), php.zRanges_.end(), [](double& n) { n *= HGCalParameters::k_ScaleFromDDD; });
-#ifdef EDM_ML_DEBUG
-  edm::LogVerbatim("HGCalGeom") << "HGCalParameters: Z-Boundary " << php.zRanges_[0] << ":" << php.zRanges_[1] << ":"
-                                << php.zRanges_[2] << ":" << php.zRanges_[3];
-#endif
+
   const auto& dummy2 = getDDDArray("LayerOffset", sv, 1);
   php.layerOffset_ = dummy2[0];
   php.layerCenter_ = dbl_to_int(DDVectorGetter::get("LayerCenter"));
+
+  loadSpecParsHexagon8(php);
+}
+
+void HGCalGeomParameters::loadSpecParsHexagon8(const cms::DDFilteredView& fv,
+                                               const cms::DDVectorsMap& vmap,
+                                               HGCalParameters& php,
+                                               const std::string& sdTag1) {
+  php.cellThickness_ = fv.get<std::vector<double> >(sdTag1.c_str(), "CellThickness");
+  std::for_each(
```

Comment:
Rather than repeating this `for_each` ~~5~~ ~~10~~ 15 times, probably better to make it an actual function that modifies the vector by reference.

Reply:
Done